rclcpp/src/rclcpp/executor.cpp

```diff
@@ -350,8 +350,7 @@ Executor::execute_subscription(rclcpp::SubscriptionBase::SharedPtr subscription)
   if (subscription->is_serialized()) {
     // This is the case where a copy of the serialized message is taken from
     // the middleware via inter-process communication.
-    std::shared_ptr<rcl_serialized_message_t> serialized_msg =
-      subscription->create_serialized_message();
+    auto serialized_msg = subscription->create_serialized_message();
```

Comment:
nit: being explicit about types is preferred, except in extremely verbose types (iterators, etc).